roles/customize_home/tasks/clone_specified_repo.yml

```diff
@@ -0,0 +1,29 @@
+- name: Ensure temporary home directory does not exist
+  file:
+    path: "{{ ansible_env.HOME }}/tmp_custom_home/"
+    state: absent
+
+- name: Clone specified repo into temporary custom home directory
+  git:
+    repo: "{{ custom_home_git_repo }}"
+    dest: "{{ ansible_env.HOME }}/tmp_custom_home/"
```

Comment:
I cloned into a temporary directory because git won't allow you to clone into a non-empty existing directory